firmware/arduino/chessboard/uart.ino

```diff
@@ -43,6 +44,14 @@ void serialEvent2()
         // Reset buffer position
         if (incomingByte  ==  '~')
         {
+            // Send message to Pi if the previous instruction was incomplete
+            if (currentState == IN_PROGRESS)
```

Comment:
This if statement will be true every time a new message comes in, since you set `currentState = IN_PROGRESS` right above this. Could maybe get rid of `IN_PROGRESS` state entirely, and just check `byte_num`
```c
if (byte_num != -1) {
   // blah
}
```